src/main/java/org/alfasoftware/soapstone/ParentAwareModelResolver.java

```diff
@@ -118,7 +122,31 @@
 
     definedTypes.putIfAbsent(_typeName(type), type);
 
-    return super.resolve(annotatedType, context, chain);
+    return context.getDefinedModels().containsKey(_typeName(type)) ?
+        context.getDefinedModels().get(_typeName(type)) :
+        super.resolve(annotatedType, context, chain);
+  }
+
+
+  /**
+   * We ensure super types are resolved first, this makes sure that inheritance ends up being properly documented
+   */
+  private void resolveSuperTypes(AnnotatedType annotatedType, ModelConverterContext context) {
+
+    Type type = annotatedType.getType();
+    JavaType javaType = objectMapper().getTypeFactory().constructType(type);
```

Comment:
is there a risk of a npe here? if objectMapper().getTypeFactory() was null...

Reply:
@willhowes - The object mapper requires a typefactory when it is instantiated, so we can safely assume it won't be null unless something has gone very wrong! :)